src/test/java/com/exasol/containers/tls/CertificateProviderTest.java

```diff
@@ -0,0 +1,196 @@
+
+import static com.exasol.testutil.ExceptionAssertions.assertThrowsWithMessage;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.when;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.exasol.config.ClusterConfiguration;
+import com.exasol.containers.ContainerFileOperations;
+import com.exasol.containers.ExasolContainerException;
+
+@ExtendWith(MockitoExtension.class)
+class CertificateProviderTest {
+
+    private static final String INVALID_CERTIFICATE_CONTENT = "invalid certificate content";
+    private static final String TLS_CERT_PATH = "tls cert path";
+    @Mock
+    private ContainerFileOperations fileOperationsMock;
+    @Mock
+    private Supplier<Optional<ClusterConfiguration>> configSupplierMock;
+    @Mock
+    private ClusterConfiguration clusterConfigMock;
+
+    private CertificateProvider certificateProvider;
+
+    @BeforeEach
+    void setup() {
+        this.certificateProvider = new CertificateProvider(this.configSupplierMock, this.fileOperationsMock);
+    }
+
+    @Test
+    void testGetCertificateSucceeds() throws ExasolContainerException {
+        simulateTlsCert(readResource("/ssl.crt"));
+
+        final Optional<X509Certificate> certificate = this.certificateProvider.getCertificate();
+
+        assertThat(certificate.isPresent(), is(true));
+        assertThat(certificate.get(), notNullValue());
+        assertThat(certificate.get().getIssuerDN().getName(), equalTo("CN=exacluster.local"));
+    }
+
+    private void simulateTlsCert(final String certContent) throws ExasolContainerException {
+        when(this.configSupplierMock.get()).thenReturn(Optional.of(this.clusterConfigMock));
+        when(this.clusterConfigMock.getTlsCertificatePath()).thenReturn(TLS_CERT_PATH);
+        when(this.fileOperationsMock.readFile(TLS_CERT_PATH, StandardCharsets.UTF_8)).thenReturn(certContent);
+    }
+
+    private String readResource(final String resourceName) {
+        try {
+            final byte[] allBytes = ((BufferedInputStream) getClass().getResource(resourceName).getContent())
+                    .readAllBytes();
+            return new String(allBytes, StandardCharsets.UTF_8);
+        } catch (final IOException exception) {
+            throw new AssertionError("Error reading resource " + resourceName, exception);
+        }
+    }
+
+    @Test
+    void testGetCertificateReturnsEmptyOptionalForMissingConfiguration() throws ExasolContainerException {
+        when(this.configSupplierMock.get()).thenReturn(Optional.empty());
+
```

Comment:
Code style: no empty lines in methods

Reply:
I added the empty lines by intention to make the test easier to read and to separate precondition, the action and assertions.

Reply:
yes. But it's our code style not to do this.

Rationale: If free lines a are required, better extract methods

Reply:
OK, done.